dartagnan/src/main/java/com/dat3m/dartagnan/encoding/WmmEncoder.java

```diff
@@ -227,17 +223,49 @@ private Map<Relation, EventGraph> initializeLazyEncodeSets(LazyRelationAnalysis
     }
 
     private final class RelationEncoder implements Constraint.Visitor<Void> {
+        private static final Set<Class<? extends Definition>> STATIC_RELATIONS = new HashSet<>(Arrays.asList(
+                ProgramOrder.class,
+                External.class,
+                Internal.class,
+                AMOPairs.class,
+                DirectControlDependency.class,
+                CASDependency.class,
+                SameInstruction.class,
+                SameScope.class,
+                SyncWith.class,
+                SameVirtualLocation.class, // FIXME?!
+                Empty.class,
+                // Static because the underlying sets are static:
+                SetIdentity.class,
+                CartesianProduct.class
+        ));
+
         final Program program = context.getTask().getProgram();
         final RelationAnalysis ra = context.getAnalysisContext().requires(RelationAnalysis.class);
         final BooleanFormulaManager bmgr = context.getBooleanFormulaManager();
         final List<BooleanFormula> enc = new ArrayList<>();
 
+        // ASSUMPTION: The encode-set of a static relation only contains
+        //             possible edges. If not, this encodes wrong semantics.
```

Comment:
I find this comment unclear. Is this what you mean by "A's may-set computation for static relations is maximally precise, i.e., it is no over-approximation."?